### PR TITLE
Filled in missing documentation in commonly used APIs

### PR DESCRIPTION
**Pull request by Joannis**
I've added documentation for some of the most commonly used Vapor APIs that were still missing doc comments.

### CODE REVIEW

File: Sources/Vapor/Application.swift [hunk 1]
```diff
@@ -94,6 +94,10 @@ public final class Application {
         DotEnvFile.load(for: environment, on: .shared(self.eventLoopGroup), fileio: self.fileio, logger: self.logger)
     }
     
+    /// Starts the Application using the `start()` command, then waits for any running tasks to complete
```
Comment (gwynne) [inline Sources/Vapor/Application.swift:97]: I would say "method" rather than "command" here.

File: Sources/Vapor/Application.swift [hunk 2]
```diff
@@ -94,6 +94,10 @@ public final class Application {
...
+    /// If your application is started without arguments, the default argument is used.
+    ///
+    /// Under normal circumstances, `run `will start the shut down and wait for the web server to (manually) shut down before exiting the application.
```
Comment (gwynne) [inline Sources/Vapor/Application.swift:100]:
```suggestion
    /// Under normal circumstances, `run() `will begin the shutdown, then wait for the web server to (manually) shut down before returning.
```
(Yes, the use of both "shut down" and "shutdown" is on purpose 😅)

File: Sources/Vapor/Request/Request.swift [hunk 3]
```diff
@@ -99,6 +99,8 @@ public final class Request: CustomStringConvertible {
         }
     }
 
+    /// This container is used to read your Decodable type using a ContentDecoder implementation.
+    /// If no ContentDecoder is provided, a supported Decoder will be selected based on the Request's Content-Type header.
```
Comment (gwynne) [inline Sources/Vapor/Request/Request.swift:103]:
```suggestion
    /// This container is used to read your `Decodable` type using a `ContentDecoder` implementation.
    /// If no `ContentDecoder` is provided, the `Request`'s `Content-Type` header is used to look up a decoder registered for that type.
```

File: Sources/Vapor/Request/Request.swift [hunk 4]
```diff
@@ -142,12 +146,21 @@ public final class Request: CustomStringConvertible {
...
     }
 
+    /// The address from which this HTTP request was received by SwiftNIO.
+    /// This address may not represent the original address of the peer, especially if Vapor receives its requests through a reverse-proxy such as nginx.
     public let remoteAddress: SocketAddress?
     
+    /// The EventLoop upon which Vapor calls the middlewares and route handler for this request.
+    /// While respond to this request asynchronously, you **MUST** ensure the resulting EventLoopFuture is bound to this EventLoop.
+    ///
+    /// If you cannot ensure that a library returns on this EventLoop, you can use `eventLoopFuture.hop(to: request.eventLoop)`
```
Comment (gwynne) [inline Sources/Vapor/Request/Request.swift:156]:
```suggestion
    /// The `EventLoop` which is handling this request. The route handler and any relevant middleware are invoked on this event loop.
    /// 
    /// - Warning: A futures-based route handler **MUST** return an `EventLoopFuture` bound to this event loop.
    ///   If this is difficult or awkward to guarantee, use `EventLoopFuture.hop(to:)` to jump to this event loop.
```

**Review by gwynne — CHANGES_REQUESTED** [review]
Just some grammatical and phrasing nits. I make no warranty that any of the suggestions are indisputably improvements in any given person's eyes, not even my own 😉.

File: Sources/Vapor/Client/Client.swift [hunk 5]
```diff
@@ -14,7 +14,7 @@ extension Client {
 extension Client {
     public func get(_ url: URI, headers: HTTPHeaders = [:], beforeSend: (inout ClientRequest) throws -> () = { _ in }) -> EventLoopFuture<ClientResponse> {
         return self.send(.GET, headers: headers, to: url, beforeSend: beforeSend)
-    }
+    }`
```
Comment (0xTim) [inline Sources/Vapor/Client/Client.swift:17]: Random backtick 
```suggestion
    }
```

File: Sources/Vapor/Application.swift (hunk 1 above)
Comment (0xTim) [reply Sources/Vapor/Application.swift:97]: Agree, we don't have a start command so that could be seen as confusing

**Review by 0xTim — CHANGES_REQUESTED** [review]
Couple of minor nits then we're good

**Review by 0xTim — APPROVED** [review]

**Pull request merged**